File: dev_stt_2019/embedded_system_surprisy/embedded_system/library_cpp/led_cpp/Led.cpp
#include "Led.h"

using namespace std;

Led::Led(int pn,string c): pin_number(pn), color(c) {
    if(wiringPiSetupGpio() == -1) { //when initialize wiringPi failed, print message to screen
        printf("setup wiringPi failed !\n");
    }
}
void Led::on() {
    pinMode(pin_number, OUTPUT);
    digitalWrite(pin_number, HIGH);   //LED on
    // printf("LED  on\n");
}
void Led::off()
{
    pinMode(pin_number, OUTPUT);
    digitalWrite(pin_number, LOW);   //LED on
    // printf("LED  off\n");
}

File: dev_stt_2019/embedded_system_surprisy/embedded_system/library_py/led_py/Led.py
#Import Library
import RPi.GPIO as GPIO

#naming convention
GPIO.setmode(GPIO.BCM)
#GPIO warning messages
GPIO.setwarnings(False)

class Led():
    def __init__(self, pin_number):
        self.name = "LED"
        self.pin_number = pin_number
        self.status = False
        GPIO.setup(self.pin_number,GPIO.OUT)
        GPIO.output(self.pin_number,GPIO.LOW)

    def on(self):
        self.status = True
        GPIO.output(self.pin_number,GPIO.HIGH)

    def off(self):
        self.status = False
        GPIO.output(self.pin_number,GPIO.LOW)
